anx/provider/loadbalancer/reconciliation/reconciliation.go

```diff
@@ -40,14 +40,14 @@ type Reconciliation interface {
 	//
 	// Some resources need others to already exist, so creating all resources returned by ReconcileCheck
 	// and then calling ReconcileCheck again will not always result in "nothing to change".
-	ReconcileCheck() (toCreate []types.Object, toDestroy []types.Object, err error)
+	ReconcileCheck(ctx context.Context) (toCreate []types.Object, toDestroy []types.Object, err error)
 
 	// Reconcile calls ReconcileCheck in a loop, each time first destroying and then creating Objects until ReconcileCheck returns no operations to be done.
 	// Once Reconcile returns without error, reconciliation is complete.
-	Reconcile() error
+	Reconcile(ctx context.Context) error
 
 	// Status returns a map with external IP address as key and array of ports as value, based on the current state in the Engine.
-	Status() (map[string][]uint16, error)
+	Status(ctx context.Context) (map[string][]uint16, error)
```

Comment:
no changes here, use context from the `reconciliation` object these are called on